docs/contract-rate-non-revision-design.md

```diff
@@ -0,0 +1,77 @@
+# Contract and Rate Without Revisions Design
+
+This is an ER diagram of the data that is planned to be sent to Salesforce on submission. It relies on Salesforce's native diff tracking so elides the concept of Revisions. 
+
+```mermaid
+erDiagram
+
+Contract {
+    String id
+    DateTime initiallySubmittedAt
+    DateTime updatedAt
+    StateCode stateCode
+    String name
+    String mccrsID
+
+    String[] programNicknames
+    PopulationCoveredEnum populationCovered "MEDICAID, CHIP, MEDICAID_AND_CHIP" 
+    SubmissionTypeEnum submissionType "CONTRACT_ONLY, CONTRACT_AND_RATES"
+    Boolean riskBasedContract
+    String submissionDescription
+    Document[] contractDocuments
+    Document[] supportingDocuments
+    ContractTypeEnum contractType "BASE, AMENDMENT"
+    ContractExecutionStatusEnum contractExecutionStatus "EXECUTED, UNEXECUTED"
+    CalendarDate contractDateStart
+    CalendarDate contractDateEnd
+    String[] managedCareEntities
+    String[] federalAuthorities
+    Boolean statutoryRegulatoryAttestation
+    String statutoryRegulatoryAttestationDescription
+
+    Boolean inLieuServicesAndSettings
+    Boolean modifiedBenefitsProvided
+    Boolean modifiedGeoAreaServed
+    Boolean modifiedMedicaidBeneficiaries
+    Boolean modifiedRiskSharingStrategy
+    Boolean modifiedIncentiveArrangements
+    Boolean modifiedWitholdAgreements
+    Boolean modifiedStateDirectedPayments
+    Boolean modifiedPassThroughPayments
+    Boolean modifiedPaymentsForMentalDiseaseInstitutions
+    Boolean modifiedMedicalLossRatioStandards
+    Boolean modifiedOtherFinancialPaymentIncentive
+    Boolean modifiedEnrollmentProcess
+    Boolean modifiedGrevienceAndAppeal
+    Boolean modifiedNetworkAdequacyStandards
+    Boolean modifiedLengthOfContract
+    Boolean modifiedNonRiskPaymentArrangements
+    Boolean statutoryRegulatoryAttestation
+    Boolean statutoryRegulatoryAttestationDescription
+    
+}
+
+Rate {
+    String id
+    DateTime initiallySubmittedAt
+    DateTime updatedAt
+    StateCode stateCode
+    String name
+    
+    RateTypeEnum rateType "NEW, AMENDMENT"
+    RateCapitationTypeEnum rateCapitationType "RATE_CELL, RATE_RANGE"
+
+    String[] programNicknames
```

Comment:
Reminder that rate programs have the logic where sometimes we display historic rate program names sometimes not. See `useHistoricPrograms` to see that logic. Seems like the mc-review API will have to do the logic and return the proper rate programs depending on submission...